runtime/src/main/java/org/corfudb/runtime/view/ManagementView.java

```diff
@@ -90,4 +113,161 @@ public void healNode(@Nonnull String endpointToHeal, int retry, @Nonnull Duratio
                          @Nonnull Duration pollPeriod) {
         new HealNode(endpointToHeal, runtime, retry, timeout, pollPeriod).invoke();
     }
+
+    /**
+     * Analyzes the health of the cluster based on the views of the cluster of all the
+     * ManagementAgents.
+     * STABLE: if all nodes in the layout are responsive.
+     * DEGRADED: if a minority of Layout servers
+     * or a minority of LogUnit servers - in QUORUM_REPLICATION mode only are unresponsive.
+     * UNAVAILABLE: if a majority of Layout servers or the Primary Sequencer
+     * or a node in the CHAIN_REPLICATION or a majority of nodes in QUORUM_REPLICATION is
+     * unresponsive.
+     *
+     * @param layout              Layout based on which the health is analyzed.
+     * @param peerResponsiveNodes Responsive nodes according to the management services.
+     * @return ClusterStatus
+     */
+    private ClusterStatus getClusterHealth(Layout layout, Set<String> peerResponsiveNodes) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)